cli/src/commands/gerrit.rs

```diff
@@ -0,0 +1,137 @@
+    /// "mailing in" something to be read by someone else.
+    Mail(MailCommand),
+}
+
+pub fn cmd_gerrit(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    subcommand: &GerritCommand,
+) -> Result<(), CommandError> {
+    match subcommand {
+        GerritCommand::Mail(review) => {
+            cmd_mail(ui, command, review)?;
+        }
+    }
+    Ok(())
+}
+
+#[derive(clap::Args, Clone, Debug)]
+pub struct MailCommand {
+    /// The revset, selecting which commits are mailed in to Gerrit. This can be
+    /// any arbitrary set of commits; they will be modified to include a
+    /// `Change-Id` footer if one does not already exist, and then mailed off to
+    /// Gerrit for review.
+    #[arg(long, short = 'r')]
+    revision: Vec<RevisionArg>,
+
+    /// The origin where your changes will be sent. This MUST be a reference of
+    /// the form `branch@remote`, where `branch` is the target branch name that
+    /// the change will be submitted to, and `remote` is the remote to push to.
+    /// This remote MUST be a Git push URL for your Gerrit instance.
+    #[arg(long = "for", short = 'f', default_value = "main@origin")]
+    for_: String,
+}
+
+pub fn cmd_mail(
+    ui: &mut Ui,
+    command: &CommandHelper,
+    mail: &MailCommand,
+) -> Result<(), CommandError> {
+    let mut workspace_command = command.workspace_helper(ui)?;
+
+    let to_mail: IndexSet<Commit> =
+        resolve_multiple_nonempty_revsets(&mail.revision, &workspace_command, ui)?;
+    if to_mail
+        .iter()
+        .any(|commit| commit.id() == workspace_command.repo().store().root_commit_id())
+    {
+        return Err(user_error("Cannot mail the root commit"));
+    }
+
+    workspace_command.check_rewritable(to_mail.iter())?;
+
+    // make sure that the --for parameter is a valid refspec of the
+    // form 'branch@remote'
+    let for_parts: Vec<&str> = mail.for_.split('@').collect();
+    if for_parts.len() != 2 {
+        return Err(user_error(
+            "The --for parameter must take the form 'branch@remote' for a given git 'remote'",
+        ));
+    }
+
+    let for_branch = for_parts[0];
+    let for_remote = for_parts[1];
+
+    let tx = workspace_command.start_transaction();
+    let base_repo = tx.base_repo().clone();
+    let store = base_repo.store();
+    let git_repo = get_git_repo(&store)?; // will fail if not a git repo
+
+    // make sure the remote exists
+    let mut remote_exists = false;
+    for remote in git_repo.remotes()?.iter() {
+        if remote == Some(for_remote) {
+            remote_exists = true;
+            break;
+        }
+    }
+
+    if !remote_exists {
+        return Err(user_error(&format!(
+            "The (Gerrit) remote '{}' does not exist",
+            for_remote
+        )));
+    }
+
+    for head in base_repo.index().heads(&mut to_mail.iter().map(|c| c.id())) {
+        println!("Pushing to {}@{}: {:?}", for_branch, for_remote, head);
```

Comment:
should use `writeln!(ui.stdout(), ...)` (or `ui.stderr()`?) instead of `println!()`

Reply:
Oh, damn. This is in the base commit but I did reword it in the follow up implementation. I think I'll probably just drop the printing in this case...

Reply:
I went ahead and squashed these two together, fixing this.